src/mrmimeparser.c

```diff
@@ -1024,8 +1027,55 @@ static int mrmimeparser_add_single_part_if_known(mrmimeparser_t* ths, struct mai
 					part = mrmimepart_new();
 					part->m_type = MR_MSG_TEXT;
 					part->m_int_mimetype = mime_type;
-					part->m_msg = simplified_txt;
-					part->m_msg_raw = strndup(decoded_data, decoded_data_bytes);
+					
+                    /* cs:
+                     * 
+                     * Not shure if  part->m_msg  is the correct string to work on (!)
+                     * Maybe part->m_msg_raw should be handled (too?)
+                     * 
+                     */
+                    
+                    part->m_msg = simplified_txt;  // (r10s) this should contain all msg text incl. uuencoded parts
+					
+                    /*
+                     *      New code from cs 
+                     *
+                     *  
+                     * The following code uses mr_... functions of mruudecode.c
+                     * 
+                     * taken from cs testing sources mruudecode.c (main)
+                     * 
+                     * here reduced to the real needs
+                     * 
+                     * uuencoded parts in part->m_msg will be
+                     *      - detected,
+                     *      - replaced by a replacement text (in the moment)
+                     *          a) needs to be adapted to the requirements here
+                     *          b) link to the generated file (!?) (todo)
+                     *      - decoded and saved to a file
+                     * 
+                     */
+                     
+                    char* lineend = "\r\n";  // default line end
+                    char* uustartpos;
+                    char* source_str = part->m_msg; // original string is changed (!)
+    
+                    lineend = mr_detect_line_end(part->m_msg);
+    
+                    while(1){
+                        /* any number of parts possible */
+                        uustartpos = mr_find_uuencoded_part(source_str, lineend);
+                        if(uustartpos){
+                            /* todo: path to created files is not handled in the moment */
+                            source_str = mr_handle_uuencoded_part (source_str, uustartpos, lineend);
+                        }
+                        else{
+                            break;
+                        }
+                    }
```

Comment:
not sure how the blob is returned can can be given to mrmimepart_new().

Reply:
In the moment only the message without the uuencoded part is finally returned. The decoded data itself is directly written to a file.